tests/vmi_networking_test.go

```diff
@@ -722,42 +722,46 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
-					ipv4NetworkCIDR = api.DefaultVMCIDR
+			for _, serverIP := range serverVMI.Status.Interfaces[0].IPs {
+				if netutils.IsIPv6String(serverIP) {
+					Skip("Skip till https://github.com/kubevirt/kubevirtci/pull/432 is merged")
 				}
-				By("Checking ping to gateway")
-				ipAddr := gatewayIPFromCIDR(ipv4NetworkCIDR)
-				Expect(tests.PingFromVMConsole(serverVMI, ipAddr)).To(Succeed())
 
-				By("Checking ping to google")
-				Expect(tests.PingFromVMConsole(serverVMI, "8.8.8.8")).To(Succeed())
-				Expect(tests.PingFromVMConsole(clientVMI, "google.com")).To(Succeed())
-			}
-
-			By("starting a tcp server")
-			tcpPort := "8080"
-			err = tests.CheckForTextExpecter(serverVMI, createExpectStartTcpServer(tcpPort), 30)
-			Expect(err).ToNot(HaveOccurred())
+				if netutils.IsIPv6String(serverIP) {
+					By("Checking traceroute from vmi to cluster nodes gateway (IPv6)")
+					// Cluster nodes subnet (docker network gateway)
+					// Docker network subnet cidr definition https://bit.ly/2wZTgMK
+					err := tests.CheckForTextExpecter(serverVMI, createExpectTraceroute6("2001:db8:1::1"), 30)
+					Expect(err).ToNot(HaveOccurred(), "Failed to traceroute to VMI %s within the given timeout", serverVMI.Name)
+				} else {
+					if ipv4NetworkCIDR == "" {
+						ipv4NetworkCIDR = api.DefaultVMCIDR
+					}
+					By("Checking ping (IPv4) to gateway")
+					ipAddr := gatewayIPFromCIDR(ipv4NetworkCIDR)
+					Expect(tests.PingFromVMConsole(serverVMI, ipAddr)).To(Succeed())
+
+					By("Checking ping (IPv4) to google")
+					Expect(tests.PingFromVMConsole(serverVMI, "8.8.8.8")).To(Succeed())
+					Expect(tests.PingFromVMConsole(clientVMI, "google.com")).To(Succeed())
+				}
 
-			By("Connecting from the client vm")
-			err = tests.CheckForTextExpecter(clientVMI, createExpectConnectToServer(serverIP, tcpPort, true), 30)
-			Expect(err).ToNot(HaveOccurred())
+				By("starting a tcp server")
+				tcpPort := "8080"
+				err = tests.CheckForTextExpecter(serverVMI, createExpectStartTcpServer(tcpPort), 30)
+				Expect(err).ToNot(HaveOccurred())
 
-			By("Rejecting the connection from the client to unregistered port")
-			err = tests.CheckForTextExpecter(clientVMI, createExpectConnectToServer(serverIP, "8081", false), 30)
-			Expect(err).ToNot(HaveOccurred())
+				By("Connecting from the client vm")
+				err = tests.CheckForTextExpecter(clientVMI, createExpectConnectToServer(serverIP, tcpPort, true), 30)
+				Expect(err).ToNot(HaveOccurred())
 
-			err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(serverVMI.Name, &v13.DeleteOptions{})
-			Expect(err).ToNot(HaveOccurred())
+				By("Rejecting the connection from the client to unregistered port")
+				err = tests.CheckForTextExpecter(clientVMI, createExpectConnectToServer(serverIP, "8081", false), 30)
+				Expect(err).ToNot(HaveOccurred())
 
+				err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(serverVMI.Name, &v13.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
This should not be in the for-loop.

Reply:
Thanks done, as i wrote i stopped checking when i saw preliminary needs.